server/src/main/java/io/crate/types/GeoShapeType.java

```diff
@@ -148,11 +149,70 @@ public StorageSupport<Map<String, Object>> storageSupport() {
         return STORAGE;
     }
 
+    private static long sizeOf(Object object) {
+        // Not Using RamUsageEstimator.sizeOfMap because it is limited to depth 1
+        // Coordinates can be arrays/lists with 4 levels,
+        // and in geometry collections the whole thing is within a list of objects
+        //
+        // depending on the source of the GeoJSON it is either using lists or double arrays
+        return switch (object) {
+            case Map<?, ?> map -> sizeOf(map);
+            case Collection<?> collection -> {
+                long bytes = RamUsageEstimator.shallowSizeOf(collection);
+                for (var item : collection) {
+                    bytes += sizeOf(item);
+                }
+                yield bytes;
+            }
+            case double[][][][] multiPolygon -> {
+                long bytes = RamUsageEstimator.NUM_BYTES_ARRAY_HEADER;
+                for (double[][][] polygon : multiPolygon) {
+                    bytes += sizeOf(polygon);
+                }
+                yield bytes;
+            }
+            case double[][][] polygon -> {
+                long bytes = RamUsageEstimator.NUM_BYTES_ARRAY_HEADER;
+                for (double[][] lineRing : polygon) {
+                    bytes += sizeOf(lineRing);
+                }
+                yield bytes;
+            }
+            case double[][] lineRing -> {
+                long bytes = RamUsageEstimator.NUM_BYTES_ARRAY_HEADER;
+                for (double[] points : lineRing) {
+                    bytes += RamUsageEstimator.sizeOf(points);
+                }
+                yield bytes;
+            }
+            default -> RamUsageEstimator.sizeOfObject(object);
+        };
+    }
+
+    private static long sizeOf(Map<?, ?> map) {
+        long bytes = RamUsageEstimator.shallowSizeOf(map);
```

Comment:
Can we re-use https://github.com/crate/crate/commit/6e2ed5b40563502cf8c8a48189166e7f60f3c96c#diff-c08ccdb6feb92821661a37bd80513b591b9898bc97ed227263ca4212be9f546eR387 (maybe by moving this into a shared `io.crate.breaker.RamUsageEstimator`)?